scripts/upgrade

```diff
@@ -11,41 +11,62 @@ source /usr/share/yunohost/helpers
 #=================================================
 # LOAD SETTINGS
 #=================================================
-ynh_script_progression --message="Loading installation settings..." --weight=2
+ynh_script_progression --message="Loading installation settings..." --weight=1
 
 app=$YNH_APP_INSTANCE_NAME
 
 domain=$(ynh_app_setting_get --app=$app --key=domain)
 path_url=$(ynh_app_setting_get --app=$app --key=path)
-is_public=$(ynh_app_setting_get --app=$app --key=is_public)
 final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 
 #=================================================
 # CHECK VERSION
 #=================================================
 
+### This helper will compare the version of the currently installed app and the version of the upstream package.
+### $upgrade_type can have 2 different values
+### - UPGRADE_APP if the upstream app version has changed
+### - UPGRADE_PACKAGE if only the YunoHost package has changed
+### ynh_check_app_version_changed will stop the upgrade if the app is up to date.
+### UPGRADE_APP should be used to upgrade the core app only if there's an upgrade to do.
 upgrade_type=$(ynh_check_app_version_changed)
 
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
-ynh_script_progression --message="Ensuring downward compatibility..."
-
-# Fix is_public as a boolean value
-if [ "$is_public" = "Yes" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=1
-	is_public=1
-elif [ "$is_public" = "No" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=0
-	is_public=0
-fi
+ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
 # If final_path doesn't exist, create it
 if [ -z "$final_path" ]; then
 	final_path=/var/www/$app
 	ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 fi
 
+
+###
+### MIGRATE FROM LEGACY PERMISSION SYSTEM
+###
+
+# Remove skipped_uris if exists
+# /!\ This commands also remove the "main" permission from "visitor" groups.
+ynh_app_setting_delete --app=$app --key=skipped_uris
+
+# Remove unprotected_uris if exists
+ynh_app_setting_delete --app=$app --key=unprotected_uris
+
+# Remove protected_uris if exists
+ynh_app_setting_delete --app=$app --key=protected_uris
+
+# Remove is_public if exists and trick about what I said 8 lines before.
+if [ -n "$is_public" ]; then
+	if [ $is_public -eq 1 ]; then
+		ynh_permission_update --permission "main" --add "visitors"
+        fi
+	ynh_app_setting_delete --app=$app --key=is_public
+fi
```

Comment:
Please amend this section with Maniack's proposal:

```bash
if [ -n "$is_public" ]; then
        ynh_app_setting_delete --app=$app --key=skipped_uris
	if [ $is_public -eq 1 ]; then
		ynh_permission_update --permission "main" --add "visitors"
        fi
	ynh_app_setting_delete --app=$app --key=is_public
fi
```

Reply:
Regarding the 

```
if [ $is_public -eq 1 ]; then
    ynh_permission_update --permission "main" --add "visitors"
fi
```

part, are you folks sure about this ?

My understanding is that, with the new permission system, the value of `is_public` does not reflect anymore the fact that the app is public or not. The app could have been turned to private (= remove visitors) from the webadmin or CLI, and this is *not* propagated to the is_public setting of the app (now realizing maybe we could ... but there will be edge cases).

So if we're re-adding visitors here, we risk exposing an app which was meant to be private, and the admin not realizing it.

That's what I attempted to explain [in the doc here](https://github.com/YunoHost/doc/blob/master/groups_and_permissions.md#migrating-away-from-the-legacy-permission-management) : 

>Applications scripts should absolutely NOT mess up with any already-existing app accesses (including unprotected/skipped_uris settings) during any other case, as it would reset any admin-defined access rule!

Reply:
As discussed on a comment above, and here https://github.com/YunoHost/example_ynh/pull/111#discussion_r400354745

Reply:
I'm writing an helper to know if the app is still really public: `ynh_permission_has_user`

Reply:
A possible solution here https://github.com/YunoHost/example_ynh/pull/112